MapleServer2/PacketHandlers/Game/RequestTaxiHandler.cs

```diff
@@ -14,62 +15,64 @@ class RequestTaxiHandler : GamePacketHandler
+            DiscoverTaxi = 0x5
+        };
+
         public override void Handle(GameSession session, PacketReader packet)
         {
-            byte mode = packet.ReadByte();
-            long price = 0;
-            bool paid = false;
+            RequestTaxiMode mode = (RequestTaxiMode)packet.ReadByte();
+
             int mapId = 0;
+            long mesoPrice = 60000;
 
-            if (mode != 5)
+            if (mode != RequestTaxiMode.DiscoverTaxi)
             {
                 mapId = packet.ReadInt();
             }
 
             switch (mode)
             {
-                case 1: // car taxi
-                    price = packet.ReadShort();
-                    if (session.Player.Mesos >= price)
-                    {
-                        session.Player.Mesos -= price;
-                        session.Send(MesosPacket.UpdateMesos(session));
-                        paid = true;
-                    }
-                    break;
-                case 3: // rotors using mesos
-                    price = 60000;
-                    if (session.Player.Mesos >= price)
+                case RequestTaxiMode.Car:
+                    mesoPrice = 5000; //For now make all car taxis cost 5k, as we don't know the formula to calculate it yet.
+                    goto case RequestTaxiMode.RotorsMeso;
+                case RequestTaxiMode.RotorsMeso:
+                    if (session.Player.Mesos >= mesoPrice)
                     {
-                        session.Player.Mesos -= price;
+                        session.Player.Mesos -= mesoPrice;
                         session.Send(MesosPacket.UpdateMesos(session));
-                        paid = true;
+                        HandleTeleport(session, mapId);
                     }
                     break;
-                case 4: // rotors using merets
-                    price = 15;
-                    if (session.Player.Merets >= price)
+                case RequestTaxiMode.RotorsMeret:
+                    if (session.Player.Merets >= 15)
                     {
-                        session.Player.Merets -= price;
+                        session.Player.Merets -= 15;
                         session.Send(MeretsPacket.UpdateMerets(session));
-                        paid = true;
+                        HandleTeleport(session, mapId);
                     }
                     break;
-                case 5: // is sent after using rotors with meret, idk why..
-                    return;
+                case RequestTaxiMode.DiscoverTaxi:
+                    //TODO: Save somewhere and load somewhere? Perhaps on login.
+                    session.Send(TaxiPacket.DiscoverTaxi(session.Player.MapId));
+                    break;
             }
+        }
 
-            if (paid)
-            {
-                MapPlayerSpawn spawn = MapEntityStorage.GetRandomPlayerSpawn(mapId);
+        private void HandleTeleport(GameSession session, int mapId)
```

Comment:
This is another method that I think should live inside `Player` so that you can call `session.Player.Warp(spawn)` elsewhere, but for now since this wasn't your code, it is fine